src/FactorGraph01.jl

```diff
@@ -1017,19 +1015,21 @@ function addFactor!(dfg::G,
                       {G <: AbstractDFG,
                        R <: Union{FunctorInferenceType, InferenceType}}
   #
+  varOrderLabels = [v.label for v=Xi]
   namestring = assembleFactorName(dfg, Xi, maxparallel=maxparallel)
-  newFactor = DFGFactor{CommonConvWrapper{R}, Symbol}(Symbol(namestring))
-  newFactor.tags = union(labels, [:FACTOR]) # TODO: And session info
-  newFactor.solvable = solvable
-  # addNewFncVertInGraph!(fgl, newvert, currid, namestring, solvable)
-  newData = setDefaultFactorNode!(dfg, newFactor, Xi, deepcopy(usrfnc), multihypo=multihypo, threadmodel=threadmodel)
+  solverData = getDefaultFactorData(dfg, Xi, deepcopy(usrfnc), multihypo=multihypo, threadmodel=threadmodel)
+  newFactor = DFGFactor(Symbol(namestring),
+                        varOrderLabels,
+                        solverData;
+                        tags=Set(union(labels, [:FACTOR])),
+                        solvable=solvable)
 
   # TODO: Need to remove this...
   for vert in Xi
-    push!(newData.fncargvID, vert.label) # vert._internalId # YUCK :/ -- Yup, this is a problem
+    push!(solverData.fncargvID, vert.label)
   end
 
-  success = DFG.addFactor!(dfg, Xi, newFactor)
+  success = DFG.addFactor!(dfg, newFactor)
```

Comment:
Note, I recommend we use this with variable order extracted from the factor.

Reply:
Perhaps I could ask for a bit more detail on this example please?  Willing to give it a try, as long as we provide both APIs until we decide which is the long term strategy.

Reply:
Both can stay forever. just recommend this one. So drivers should implement this one, the others are all done for AbstractDFG.

Reply:
see here: https://github.com/JuliaRobotics/DistributedFactorGraphs.jl/blob/3ec2484a2b22a00b47027a310e284ff674f08a9c/src/services/AbstractDFG.jl#L217-L236